Feature: BB-244 move notification failed processing logic to replication status processor

welansari opened the pull request:
Issue: [BB-244](https://scality.atlassian.net/browse/BB-244)

This refactoring is needed for multiple reasons:
- Send “replication failed” notification only after CRR replay has failed
- Avoid duplicating “resources” (like NotificationConfigManager) in each instance of QueueProcessor
- Keep all status processor in a single place

Commit:
feature: BB-244 fix notification destination host and port format

Commit:
feature: BB-244 remove failedReplication notification logic from queueProcessor

Commit:
feature: BB-244 add failedReplication notification logic to ReplicationStatusProcessor

bert-e commented:
# Hello kerkesni,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/backbeat/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue BB-244 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.4.20`


Please check the `Fix Version/s` of BB-244, or the target
branch of this pull request.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand commented on extensions/replication/tasks/UpdateReplicationStatus.js:
```diff
@@ -102,6 +115,9 @@ class UpdateReplicationStatus extends BackbeatTask {
             score,
         };
         this.failedCRRProducer.publishFailedCRREntry(JSON.stringify(message));
+        if (this.bucketNotificationConfig) {
```
this function may not be called, e.g. if `repConfig.monitorReplicationFailures` is not set : but we should still post a bucket notification in that case.

francoisferrand commented on tests/unit/replication/ReplicateObject.spec.js:
```diff
@@ -1,166 +0,0 @@
-const assert = require('assert');
```
we should port these tests to some `tests/unit/replication/UpdateReplicationStatus.js` ?

francoisferrand committed:
Always post replication failure notif

The message must be posted as long as BucketNotification is enabled,
even if monitorReplicationFailures is disabled: thus it must be moved
out of _pushFailedEntry()

Issue: BB-244

francoisferrand committed:
Add unit tests for UpdateReplicationStatus

Issue: BB-244

francoisferrand committed:
Use GHA for caching images

This improves caching performance, and reduces build time.

Issue: BB-244

williamlardier reviewed (approved):
LGTM! Can we mention in the ticket that the `tests` workflow was updated with GHA caching? Just to keep track of why it was done in this PR

francoisferrand commented:
/bypass_author_approval

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/8.4`



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/8.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **bypass_author_approval**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.4`



The following branches have **NOT** changed:

* `development/7.10`
* `development/7.4`
* `development/8.3`



Please check the status of the associated issue BB-244.

Goodbye kerkesni.
